### PR TITLE
add parameter u

### LINKED ISSUE #2: Parameter -u

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #2

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,15 +3,35 @@ package main
...
 	"os"
 )
 
 func main() {
+	uniqueParameter := flag.Bool("u", false, "allows you to print only unique lines.")
+	flag.Parse()
+
 	buf := bufio.NewReader(os.Stdin)
-	var checkEquality []byte
+	checkEquality := []byte("")
```
Comment (Al2Klimov) [inline main.go:17]: What's this checkEquality change for?

File: main.go [hunk 2]
```diff
@@ -3,15 +3,35 @@ package main
...
+		}
 
+		uniqueContentSplit := bytes.Split(uniqueContent, []byte("\n"))
+
+		for i := range uniqueContentSplit {
+			if bytes.Compare(uniqueContentSplit[i], uniqueContentSplit[i+1]) != 0 {
+				_, _ = os.Stdout.Write(uniqueContentSplit[i])
+			}
+		}
+	} */
```
Comment (Al2Klimov) [inline main.go]: ...

File: main.go [hunk 3]
```diff
@@ -23,14 +43,32 @@ func main() {
...
+		} else {
+			if bytes.Compare(checkEquality, content) != 0 {
+				_, _ = os.Stdout.Write(content)
+				if readErr == io.EOF {
+					break
+				}
+			}
 		}
-
+		test = checkEquality
```
Comment (Al2Klimov) [inline main.go]: Why don't you check for EOF in all cases now?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (justinmueller05082000) [reply main.go:17]: i had to change it so i can use it later as an if condition

Comment (Al2Klimov) [reply main.go:17]: Really? Does the definition matter in you case? Please verify.

Comment (justinmueller05082000) [reply main.go:17]: yes it does becuase i cant compare checkEquality properly if its not initialized in the first iteration of my for loop

File: main.go (hunk 3 above)
Comment (Al2Klimov) [reply main.go]: And this one?

https://github.com/Al2Klimov/trainee-go-uniq/pull/12/files#diff-7ddfb3e035b42cd70649cc33393fe32cR36

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```
go run . -u <<EOF
a
a
a
a
EOF
```

Doesn't work as expected.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Why don't you check for EOF in all cases now?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/add-parameter-u-2`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toUniq.txt:**

```
Dog
Dog
Rat
Cat
Dog
Cat
Cat
```

**Go code:**

**Terminal:** `go run . -u < toUniq.txt `

**Output:**

```
Rat
Cat
Dog
```

**Uniq command:**

**Terminal:**  `uniq -u < toUniq.txt `

**Output:**

```
Rat
Cat
Dog
```


**Conclusion:**
Both results are euqal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**